cpp/src/aztec/honk/pcs/ipa/ipa.test.cpp

```diff
@@ -4,50 +4,51 @@
+#include "honk/pcs/commitment_key.hpp"
+#include "honk/pcs/commitment_key.test.hpp"
 using namespace barretenberg;
+namespace honk::pcs::ipa {
 
-TEST(honk_commitment_scheme, ipa_commit)
-{
-    constexpr size_t n = 1024;
-    std::vector<barretenberg::fr> scalars(n);
-    std::vector<barretenberg::g1::affine_element> points(n);
+template <class Params> class IpaCommitmentTest : public CommitmentTest<Params> {
+    using Fr = typename Params::Fr;
+    using element = typename Params::Commitment;
+    using affine_element = typename Params::C;
+    using CK = typename Params::CK;
+    using VK = typename Params::VK;
+    using Polynomial = barretenberg::Polynomial<Fr>;
 
-    for (size_t i = 0; i < n; i++) {
-        scalars[i] = barretenberg::fr::random_element();
-        points[i] = barretenberg::g1::affine_element(barretenberg::g1::element::random_element());
-    }
+  public:
+};
 
-    barretenberg::g1::element expected = points[0] * scalars[0];
+TYPED_TEST_SUITE(IpaCommitmentTest, CommitmentSchemeParams);
+
+TYPED_TEST(IpaCommitmentTest, ipa_commit)
+{
+    constexpr size_t n = 128;
+    auto poly = this->random_polynomial(n);
+    using newIpa = InnerProductArgument<TypeParam>;
+    auto CK = this->ck();
+    barretenberg::g1::element commitment = newIpa::commit(CK, poly);
+    auto srs_elements = CK->srs.get_monomial_points();
+    barretenberg::g1::element expected = srs_elements[0] * poly[0];
     for (size_t i = 1; i < n; i++) {
-        expected += points[i] * scalars[i];
+        expected += srs_elements[i] * poly[i];
     }
-
-    InnerProductArgument<barretenberg::fr, barretenberg::fq, barretenberg::g1> newIpa;
-    barretenberg::g1::element result = newIpa.commit(scalars, n, points);
-    EXPECT_EQ(expected.normalize(), result.normalize());
+    EXPECT_EQ(expected.normalize(), commitment.normalize());
 }
 
-TEST(honk_commitment_scheme, ipa_open)
+TYPED_TEST(IpaCommitmentTest, ipa_open)
 {
-    // generate a random polynomial coeff, degree needs to be a power of two
-    size_t n = 1024;
-    std::vector<barretenberg::fr> coeffs(n);
-    for (size_t i = 0; i < n; ++i) {
-        coeffs[i] = barretenberg::fr::random_element();
-    }
-    // generate random evaluation point x and the evaluation
-    auto x = barretenberg::fr::random_element();
-    auto eval = polynomial_arithmetic::evaluate(&coeffs[0], x, n);
-    // generate G_vec for testing, bypassing srs
-    std::vector<barretenberg::g1::affine_element> G_vec(n);
-    for (size_t i = 0; i < n; ++i) {
-        auto scalar = fr::random_element();
-        G_vec[i] = barretenberg::g1::affine_element(barretenberg::g1::one * scalar);
-    }
-    InnerProductArgument<barretenberg::fr, barretenberg::fq, barretenberg::g1> newIpa;
-    auto C = newIpa.commit(coeffs, n, G_vec);
-    InnerProductArgument<barretenberg::fr, barretenberg::fq, barretenberg::g1>::IpaPubInput pub_input;
-    pub_input.commitment = C;
+    using newIpa = InnerProductArgument<TypeParam>;
```

Comment:
We don't really use this `new*` convention anywhere except, I see, in commitment_scheme.test.cpp. Will you just call it `IPA` or something like this? Similarly, we don't use camelCase, but CamelCase or snake_case.

Reply:
Thanks. Done.